src/topic.rs

```diff
@@ -1,21 +1,36 @@
+use smallstr::SmallString;
+
 /// A message queue topic name to which messages can be published
-#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash, Default)]
-pub struct Topic(&'static str);
+// A survey of common topics found lengths between 16 and 35 bytes
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct Topic(SmallString<[u8; 36]>);
```

Comment:
I did run into this problem, since I am creating topic names dynamically - but worked around it by leaking the strings and then using a hashmap to cache the mapping from (site id, prio) -> &'static str

Reply:
I guess there's a reasonable bound on that combination, so that's not the worst. But we can do better!